diff --git a/node_modules/emittery/index.d.ts b/node_modules/emittery/index.d.ts
index 690f3c7..bca8679 100644
--- a/node_modules/emittery/index.d.ts
+++ b/node_modules/emittery/index.d.ts
@@ -82,6 +82,8 @@ declare class Emittery {
 	*/
 	static readonly listenerRemoved: unique symbol;
 
+	init(){}
+
 	/**
 	Subscribe to one or more events.
 
diff --git a/node_modules/emittery/index.js b/node_modules/emittery/index.js
index d3f9bb3..e81e7a5 100644
--- a/node_modules/emittery/index.js
+++ b/node_modules/emittery/index.js
@@ -190,6 +190,10 @@ class Emittery {
 	}
 
 	constructor() {
+		this.init();
+	}
+
+	init() {
 		anyMap.set(this, new Set());
 		eventsMap.set(this, new Map());
 		producersMap.set(this, new Map());
