diff --git a/node_modules/@ethereumjs/block/src/block.ts b/node_modules/@ethereumjs/block/src/block.ts
index 6f77530..cdde6b0 100644
--- a/node_modules/@ethereumjs/block/src/block.ts
+++ b/node_modules/@ethereumjs/block/src/block.ts
@@ -14,7 +14,14 @@ import {
   ssz,
 } from '@ethereumjs/util'
 import { keccak256 } from 'ethereum-cryptography/keccak'
-import { ethers } from 'ethers'
+// we aren't actually using ethers, so we don't want to have to add it
+// as a bundledPackage in the api-extractor.json to get the types to work.
+// instead, we're just faking the types here.
+declare namespace ethers  {
+  export namespace providers {
+    export type JsonRpcProvider = any
+  }
+}
 
 import { blockFromRpc } from './from-rpc'
 import { BlockHeader } from './header'
